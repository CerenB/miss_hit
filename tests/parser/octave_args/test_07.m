% This test is extracted from the GNU Octave testsuite
% For a full copyright please refert to args.tst from GNU Octave

% varargin and varargout with one output
function [x, varargout] = f (varargin)
  assert (nargin == 0);
  assert (nargout == 1);
  x = 2;
end

function test()
 assert (f () == 2);
end
