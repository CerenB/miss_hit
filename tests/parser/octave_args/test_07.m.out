----------------------------------------------------------------------
--  Parse tree for test_07.m
Root: Function_File
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <f>
      Inputs: Identifier <varargin>
      Outputs: Identifier <x>
      Outputs: Identifier <varargout>
    Body: Sequence_Of_Statements
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Identifier <nargin>
            RHS: Number_Literal <0>
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Identifier <nargout>
            RHS: Number_Literal <1>
      Statements: Simple_Assignment_Statement
        LHS: Identifier <x>
        RHS: Number_Literal <2>
  Functions: Function_Definition
    Signature: Function_Signature
      Name: Identifier <test>
    Body: Sequence_Of_Statements
      Statements: Naked_Expression_Statement
        Expression: Reference
          Name: Identifier <assert>
          Arguments: Binary_Operation ==
            LHS: Reference
              Name: Identifier <f>
            RHS: Number_Literal <2>
----------------------------------------------------------------------
In test_07.m, line 5
| function [x, varargout] = f (varargin)
|          ^ info: this token is not linked to the ast
In test_07.m, line 5
| function [x, varargout] = f (varargin)
|            ^ info: this token is not linked to the ast
In test_07.m, line 5
| function [x, varargout] = f (varargin)
|                       ^ info: this token is not linked to the ast
In test_07.m, line 5
| function [x, varargout] = f (varargin)
|                         ^ info: this token is not linked to the ast
In test_07.m, line 5
| function [x, varargout] = f (varargin)
|                             ^ info: this token is not linked to the ast
In test_07.m, line 5
| function [x, varargout] = f (varargin)
|                                      ^ info: this token is not linked to the ast
In test_07.m, line 6
|   assert (nargin == 0);
|                       ^ info: this token is not linked to the ast
In test_07.m, line 7
|   assert (nargout == 1);
|                        ^ info: this token is not linked to the ast
In test_07.m, line 8
|   x = 2;
|        ^ info: this token is not linked to the ast
In test_07.m, line 9
| end
| ^^^ info: this token is not linked to the ast
In test_07.m, line 11
| function test()
|              ^ info: this token is not linked to the ast
In test_07.m, line 11
| function test()
|               ^ info: this token is not linked to the ast
In test_07.m, line 12
|  assert (f () == 2);
|                    ^ info: this token is not linked to the ast
In test_07.m, line 13
| end
| ^^^ info: this token is not linked to the ast
MISS_HIT Summary: 1 file(s) analysed, everything seemes fine
